pandas/tests/indexes/test_base.py

```diff
@@ -1142,19 +1106,21 @@ def test_format(self):
             expected = [str(index[0])]
             assert formatted == expected
 
+        self.strIndex[:0].format()
+
+    def test_format_missing(self, nulls_fixture):
         # 2845
-        index = Index([1, 2.0 + 3.0j, np.nan])
+        index = Index([1, 2.0 + 3.0j, nulls_fixture])
         formatted = index.format()
         expected = [str(index[0]), str(index[1]), u('NaN')]
         assert formatted == expected
 
-        # is this really allowed?
-        index = Index([1, 2.0 + 3.0j, None])
-        formatted = index.format()
-        expected = [str(index[0]), str(index[1]), u('NaN')]
-        assert formatted == expected
+    def test_format_missing_obj(self, nulls_fixture):
```

Comment:
Trying to balance too many changes but am leaning towards combining this with the above test. Both are pretty similar but leave strange gaps in what they test across the float and object types

Reply:
sure